### PR TITLE
[Site design refresh] Update control (Button, Input, Select) sizing and colors

### PR DESCRIPTION
**Pull request by langermank**
## Summary

This PR addresses a number of control related design updates for the site design refactor, including some minor visual bugs and inconsistencies. 

[Storybook → ](https://primer-26e42269a9-26139705.drafts.github.io/storybook/?path=/story/components-accordion--default)

### Form control sizing

Form controls (input, select) sizes differed from `Button` sizes. Form controls should be the same size to ensure horizontal alignment with a submit button.

- Use `control` tokens for padding and height for `Input` and `Select`
- `medium` size is 48px in height, and `large` is 60px in height

**Large inputs:**
large inputs

**Medium inputs:**
medium inputs

### Button variant design refresh

- Update sizes (height) and inline padding
- Create optical alignment with the `arrow` variant
- Adjust colors per spec in Figma with some minor deviations in state to align closer to Primer (we use a 20% opacity for hover, and 40% for active in most cases)
- Adjust border-radius for `large` buttons

**Sizes:**
button sizes

| primary | secondary | subtle |
| -- | -- | -- |
| primary button hover state | secondary button hover state | subtle button hover state |

### Design tokens

- Introduce new button tokens that align with our updated naming convention (this is just less tech debt to clean up later for.. future me)
- Adjust `control` sizes

## What should reviewers focus on?

- Should I delete and deprecate the unused design tokens, or leave them in for now assuming we'll do a mass deprecation strategy later?
- What did I break?
- Check the visual snapshots and Storybook closely 🙌 in multiple browsers

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/2478
-

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/design-tokens/src/tokens/functional/colors/global.json [hunk 1]
```diff
@@ -103,7 +103,7 @@
       },
       "border": {
         "default": {
-          "value": "var(--base-color-scale-gray-2)",
+          "value": "var(--base-color-scale-gray-3)",
```
Comment (langermank) [inline packages/design-tokens/src/tokens/functional/colors/global.json:106]: This is set as `3` in Figma, so aligning it here.. but maybe need to update some borders to use `muted` elsewhere?

File: packages/react/src/Button/Button.module.css [hunk 2]
```diff
@@ -41,70 +52,119 @@
...
+.Button--secondary .Button__icon-visual,
+.Button--secondary .Button-arrow {
+  color: var(--brand-button-secondary-fgColor-rest);
 }
 
-.Button--label-secondary.Button-label--disabled,
-.Button--label-secondary.Button-label--disabled,
-.Button__icon-visual--disabled {
-  color: var(--brand-Button-fg-secondary-disabled);
+.Button--secondary:disabled .Button-arrow--disabled,
```
Comment (langermank) [inline packages/react/src/Button/Button.module.css:130]: I just need to say the `disabled` state situation with this component is absolutely bananas 🍌

File: packages/design-tokens/src/tokens/functional/components/button/colors.js [hunk 3]
```diff
@@ -63,5 +63,127 @@ module.exports = {
         },
       },
     },
+    button: {
```
Comment (rezrah) [inline packages/design-tokens/src/tokens/functional/components/button/colors.js:66]: Noticed this is lowercase here, so assuming this related to the the new naming convention being proposed where components / patterns aren't PascalCase?

Also, should we remove all the lines above this as those `--brand-Button-etc` variables are probably redundant now?

File: packages/react/src/ActionMenu/ActionMenu.visual.spec.ts-snapshots/Visual-Comparison-ActionMenu-ActionMenu-Disabled-Item-1-linux.png
Comment (rezrah) [inline packages/react/src/ActionMenu/ActionMenu.visual.spec.ts-snapshots/Visual-Comparison-ActionMenu-ActionMenu-Disabled-Item-1-linux.png:1]: Is the background color change intentional? I think only the menu item should change in this example.

Screenshot 2023-08-17 at 12 54 08

File: packages/react/src/Button/Button.module.css [hunk 4]
```diff
@@ -41,70 +52,119 @@
...
+  color: var(--brand-button-secondary-fgColor-disabled);
 }
 
+/* subtle */
+
 .Button--subtle {
   color: var(--brand-color-text-default);
-  transition: box-shadow 200ms;
+  border-color: var(--brand-button-subtle-borderColor-rest);
+  transition: background-color, border-color 200ms;
```
Comment (rezrah) [inline packages/react/src/Button/Button.module.css:145]: We have an equivalent token for the timing..

```suggestion
  transition: background-color, border-color var(--brand-animation-duration-faster);
```

File: packages/react/src/Button/Button.module.css [hunk 5]
```diff
@@ -41,70 +52,119 @@
...
 }
 
 .Button--subtle:hover:not(.Button[disabled]) {
-  box-shadow: var(--brand-Button-shadow-subtle-hover);
+  border-color: var(--brand-button-subtle-borderColor-hover);
+}
+
+.Button--label-subtle {
+  color: var(--brand-button-sungle-fgColor-rest);
```
Comment (rezrah) [inline packages/react/src/Button/Button.module.css:153]:
```suggestion
  color: var(--brand-button-subtle-fgColor-rest);
``

File: packages/react/src/Button/Button.visual.spec.ts-snapshots/Visual-Comparison-Button-Button-Large-1-linux.png
Comment (rezrah) [inline packages/react/src/Button/Button.visual.spec.ts-snapshots/Visual-Comparison-Button-Button-Large-1-linux.png:1]: The chevron size feels a little small here, but I'm not 100% sure what it should be...

Screenshot 2023-08-17 at 13 03 11

**Comment by jesussandreas** [comment]
my only comment would be that the padding on the `primary` button is set as `14px` instead of `16px. or maybe it's being added on inside the text?

Screenshot 2023-08-17 at 12 18 24 PM

and also the icons on the large version seems it's not scaling? 

Screenshot 2023-08-17 at 12 18 08 PM

File: packages/react/src/Button/Button.module.css (hunk 5 above)
Comment (langermank) [reply packages/react/src/Button/Button.module.css:153]: sungle 😂 

This is why we need stylelint... jus sayin' 😉

File: packages/design-tokens/src/tokens/functional/components/button/colors.js (hunk 3 above)
Comment (langermank) [reply packages/design-tokens/src/tokens/functional/components/button/colors.js:66]: Yes, exactly.

Yeah I mentioned deprecation in the PR description as I'm not sure.. if I delete them and they are used outside this repo, breaking change. What do you think?

**Comment by langermank** [comment]
@jesussandreas I think the line box height for the Button label is 16px, so yeah 16px for block padding * 2 + 16 for the text block = 48 (for medium). Regardless, we set an explicit `height` on buttons since they should never have a line-break 😄 

I also adjusted all button visuals for the large size to 20px height/width

primary button size large

**Review by jesussandreas — APPROVED** [review]

**Comment by jesussandreas** [comment]
LGTM 🚀

**Comment by jesussandreas** [comment]
@langermank this is optional but i think we should make the call on removing those super thin borders on primary buttons. it's doesn't really have a true purpose other than visual and even that is not very clear.

File: packages/design-tokens/src/tokens/functional/components/button/colors.js (hunk 3 above)
Comment (rezrah) [reply packages/design-tokens/src/tokens/functional/components/button/colors.js:66]: I'd go ahead and remove them. We can call out in a changeset which tokens have been replaced. You should be able to just take the diff from the PR comment and paste it into the changeset. WDYT?

File: packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Playground-1-linux.png
Comment (rezrah) [inline packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Playground-1-linux.png:1]: Looks like a regression on this one, with the secondary CTA overflowing the container. 

Screenshot 2023-08-25 at 12 30 25

File: packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Search-Open-1-linux.png
Comment (rezrah) [inline packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Search-Open-1-linux.png:1]: The search field in the nav bar seems a little too large now. Is it possible to use the smaller size control here? 

Screenshot 2023-08-25 at 12 30 56

**Review by rezrah — COMMENTED** [review]
This is looking great ✨ 

FYI we'll need to also make the same changes to buttons, form inputs and action menu in Figma. @langermank - are you planning to do this too? If not, we should probably have a tracking issue to account for that and the updates to interface guidelines also.

The only thing I wanted to flag, was the regression to SubdomainNavBar. Would be great to fix that before we ship this. Let me know if you'd like a hand with it.

cc. @danielguillan @josepmartins too for another set of 👀

File: packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Playground-1-linux.png
Comment (langermank) [reply packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Playground-1-linux.png:1]: Addressed 👍

File: packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Search-Open-1-linux.png
Comment (langermank) [reply packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Search-Open-1-linux.png:1]: Changed this to use `medium` size

**Comment by langermank** [comment]
@rezrah @danielguillan @josepmartins I didn't have time to get this into Figma before my break, but I'm happy to do it when I return!

I think this PR should be good to merge 🤞 I put a `minor` changeset on it but not sure if we're doing `major`.

**Review by josepmartins — APPROVED** [review]
this LGTM! great work @langermank ✨

**Comment by josepmartins** [comment]
To align to the Button component, shouldn't the rest of form elements (like the Text Input or Select) use `8px` border radius value too? Curious if we need to update our medium value to 8px vs to keeping the 6px and the 8px.

cc @jesussandreas

**Review by rezrah — APPROVED** [review]
Tests are passing ✅, this is good to go 🚀

**Pull request merged**